homework/index.js

```diff
@@ -1,46 +1,165 @@
       if (key === 'text') {
-        elem.textContent = value;
+        child.textContent = value;
       } else {
-        elem.setAttribute(key, value);
+        child.setAttribute(key, value);
       }
     });
-    return elem;
+    return child;
+  }
+  // defined for error handling
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, {
+      text: error.message,
+      class: 'alert',
+    });
+  }
+  // for creating select element
+  function createOptionElements(repositories, select) {
+    createAndAppend('option', select, {
+      text: 'Select a repository',
+      disabled: 'disabled',
+    });
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', select, {
+          text: repository.name,
+          value: index,
+        });
+      });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderRepositories(repository) {
+    const leftBox = document.getElementById('left-box');
+    leftBox.innerHTML = '';
+    const rTable = createAndAppend('table', leftBox);
+    const tbody = createAndAppend('tbody', rTable, {
+      class: 'table_elms',
+    });
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository      :',
+    });
+    createAndAppend('td', tr1, {
+      text: repository.name,
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description    :',
+    });
+    createAndAppend('td', tr2, {
+      text: repository.description,
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks          :',
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks,
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated        :',
+    });
+    createAndAppend('td', tr4, {
+      text: new Date(repository.updated_at).toLocaleString('en-GB'),
+    });
+  }
+
+  function renderContributions(repository) {
+    const rightBox = document.getElementById('right-box');
+    rightBox.innerHTML = '';
+    fetchJSON(repository.contributors_url, (error, contributions) => {
+      if (error) {
+        renderError(error);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      createAndAppend('h2', rightBox, {
+        text: 'Contributions',
+        class: 'contrib-title',
+      });
+      const contributionsList = createAndAppend('ul', rightBox, {
+        class: 'list-container',
+      });
+      contributions.forEach(contribution => {
+        const listItem = createAndAppend('li', contributionsList, {
```

Comment:
The assignment requires these list elements to be a link that opens a new tab to the GitHub page of the contributor.